### PR TITLE
Add staging and testnet configurations

### PR DESCRIPTION
**Pull request by oneEdoubleD**
I've added the configuration files for staging and testnet. I had to re-create these from .nix files (which had seperated them into multiple files) so will need to run each of these to double check they work okay. @dorin100 - as you're setting up a node testing env do you mind checking them out and seeing if they run correctly? I'll try them out too :)

### CODE REVIEW

File: configuration/defaults/testnet/configuration.yaml [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  - scKind: StdoutSK
+    scName: stdout
+    scFormat: ScText
+    scRotation: null
+
+# For the Katip logging backend this specifies the default scribes that trace
+# output is sent to if it is not configured to be sent to other scribes.
+defaultScribes:
+  - - FileSK
+    - "logs/mainnet.log"
```
Comment (ArturWieczorek) [inline configuration/defaults/testnet/configuration.yaml:108]: This should be `testnet.log`

File: configuration/defaults/staging/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  - scKind: StdoutSK
+    scName: stdout
+    scFormat: ScText
+    scRotation: null
+
+# For the Katip logging backend this specifies the default scribes that trace
+# output is sent to if it is not configured to be sent to other scribes.
+defaultScribes:
+  - - FileSK
+    - "logs/mainnet.log"
```
Comment (ArturWieczorek) [inline configuration/defaults/staging/configuration.yaml:108]: This should be `staging.log`

Comment (dorin100) [reply configuration/defaults/staging/configuration.yaml:108]: The Staging and Testnet files from this PR are identical with the ones from mainnet so they are not correct. Unfortunately, we/I don't know what should be the specific changes for Staging/Testnet.

**Comment by ArturWieczorek** [comment]
So quick question about auto generated configs with nix when using: @disassembler , @jbgi 

`nix-build -A scripts.mainnet.node -o mainnet-node-local `

The generated output config contains entries that are not present in https://github.com/input-output-hk/cardano-node/blob/master/configuration/defaults/mainnet/configuration.yaml , like:

```
  "Burst":0,
  "CacheParameter":500,
  "CriticalCQ":0.654321,
  "CriticalCQBootstrap":0.8888,
  "CriticalForkThreshold":3,
  "Enabled":false,
  "FixedTimeCQ":3600,
  "GenesisFile":"/nix/store/qd60rqsk0rjdrckg2js8dgl9jh21bd1j-mainnet-genesis.json",
  "HandshakeTimeout":30000,
  "MemPoolLimitTx":200,
  "MessageCacheTimeout":30,
  "NetworkConnectionTimeout":15000,
  "NetworkDiameter":18,
  "NodeHostAddress":"",
  "NodeId":0,
  "NodePort":3001,
  "NonCriticalCQ":0.8,
  "NonCriticalCQBootstrap":0.95,
  "NumCoreNodes":7,
  "Period":"",
  "ResponseTimeout":30000000,
  "Servers":[
    "0.pool.ntp.org",
    "2.pool.ntp.org",
    "3.pool.ntp.org"
  ],
  "SlotLength":20000,
  "StreamWindow":2048
```

**So the question is** - should those auto generated files contain those extra info or not ?
I can see that those entries were removed here: https://github.com/input-output-hk/iohk-nix/pull/368/files

Here is a full content of such config file created after building node with nix:

`cat /nix/store/w3p10f8ysrnhvrkgw3z92h23kknjf91n-config-0.json`

> {"ApplicationName":"cardano-sl","ApplicationVersion":2,"AssetLockedSrcAddress":[],"Burst":0,"CacheParameter":500,"CriticalCQ":0.654321,"CriticalCQBootstrap":0.8888,"CriticalForkThreshold":3,"Enabled":false,"FixedTimeCQ":3600,"GenesisFile":"/nix/store/qd60rqsk0rjdrckg2js8dgl9jh21bd1j-mainnet-genesis.json","HandshakeTimeout":30000,"LastKnownBlockVersion-Alt":0,"LastKnownBlockVersion-Major":1,"LastKnownBlockVersion-Minor":0,"MemPoolLimitTx":200,"MessageCacheTimeout":30,"NetworkConnectionTimeout":15000,"NetworkDiameter":18,"NodeHostAddress":"","NodeId":0,"NodePort":3001,"NonCriticalCQ":0.8,"NonCriticalCQBootstrap":0.95,"NumCoreNodes":1,"Period":"","PollDelay":1800000000,"Protocol":"RealPBFT","Rate":0,"RecoveryHeadersMessage":2200,"RequiresNetworkMagic":"RequiresNoMagic","ResponseTimeout":30000000,"Servers":["0.pool.ntp.org","2.pool.ntp.org","3.pool.ntp.org"],"SlotLength":20000,"StreamWindow":2048,"TraceBlockFetchClient":false,"TraceBlockFetchDecisions":true,"TraceBlockFetchProtocol":false,"TraceBlockFetchProtocolSerialised":false,"TraceBlockFetchServer":false,"TraceChainDb":true,"TraceChainSyncBlockServer":false,"TraceChainSyncClient":false,"TraceChainSyncHeaderServer":false,"TraceChainSyncProtocol":false,"TraceDNSResolver":false,"TraceDNSSubscription":false,"TraceErrorPolicy":false,"TraceForge":true,"TraceIpSubscription":false,"TraceLocalChainSyncProtocol":false,"TraceLocalTxSubmissionProtocol":false,"TraceLocalTxSubmissionServer":false,"TraceMempool":true,"TraceMux":false,"TraceTxInbound":false,"TraceTxOutbound":false,"TraceTxSubmissionProtocol":false,"TracingVerbosity":"NormalVerbosity","TurnOnLogMetrics":true,"TurnOnLogging":true,"ViewMode":"SimpleView","defaultBackends":["KatipBK"],"defaultScribes":[["StdoutSK","stdout"]],"hasEKG":12788,"hasPrometheus":["127.0.0.1",12798],"minSeverity":"Info","options":{"cfokey":{"value":"Release-1.0.0"},"mapBackends":{"cardano.node-metrics":["EKGViewBK",{"kind":"UserDefinedBK","name":"LiveViewBackend"}],"cardano.node.BlockFetchDecision.peers":["EKGViewBK",{"kind":"UserDefinedBK","name":"LiveViewBackend"}],"cardano.node.ChainDB.metrics":["EKGViewBK",{"kind":"UserDefinedBK","name":"LiveViewBackend"}]},"mapSubtrace":{"#ekgview":{"contents":[[{"contents":"cardano.epoch-validation.benchmark","tag":"Contains"},[{"contents":".monoclock.basic.","tag":"Contains"}]],[{"contents":"cardano.epoch-validation.benchmark","tag":"Contains"},[{"contents":"diff.RTS.cpuNs.timed.","tag":"Contains"}]],[{"contents":"#ekgview.#aggregation.cardano.epoch-validation.benchmark","tag":"StartsWith"},[{"contents":"diff.RTS.gcNum.timed.","tag":"Contains"}]]],"subtrace":"FilterTrace"},"benchmark":{"contents":["GhcRtsStats","MonotonicClock"],"subtrace":"ObservableTrace"},"cardano.epoch-validation.utxo-stats":{"subtrace":"NoTrace"},"cardano.node-metrics":{"subtrace":"Neutral"}}},"rotation":{"rpKeepFilesNum":10,"rpLogLimitBytes":5000000,"rpMaxAgeHours":24},"setupBackends":["KatipBK"],"setupScribes":[{"scFormat":"ScText","scKind":"StdoutSK","scName":"stdout","scRotation":null}]}

File: configuration/defaults/staging/configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+SocketPath: db/node.socket
+
+
+##### Core protocol parameters #####
+
+# This is the instance of the Ouroboros family that we are running.
+# The node also supports various test and mock instances.
+# "RealPBFT" is the real (ie not mock) (permissive) OBFT protocol, which
+# is what we use on mainnet in Byron era.
+Protocol = "RealPBFT";
```
Comment (ArturWieczorek) [inline configuration/defaults/staging/configuration.yaml:19]: This is not YAML format ;)

**Comment by jbgi** [comment]
> So quick question about auto generated configs with nix when using: @disassembler , @jbgi
> 
> `nix-build -A scripts.mainnet.node -o mainnet-node-local `
> 
> The generated output config contains entries that are not present in https://github.com/input-output-hk/cardano-node/blob/master/configuration/defaults/mainnet/configuration.yaml

@ArturWieczorek  We should update dependency on `iohk-nix` to use [latest config files](https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/mainnet-config.nix). This can be done with
`nix-shell -A devops --run "niv update iohk-nix"`

**Comment by dorin100** [comment]
@oneEdoubleD , @ArturWieczorek it seems that DevOps team already provides updated configs for Staging, Testnet, Mainnet. I think we need to avoid having 2 different files that would need to be maintained. 

Should we close this PR and use the configurations provided/maintained by devops? 
- staging: https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/staging-config.nix
- testnet: https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/testnet-config.nix
- mainnet: https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/mainnet-config.nix

**Comment by dorin100** [comment]
> `nix-shell -A devops --run "niv update iohk-nix"`

@jbgi thank you for the above details.

I understand that in order to start a Byron node on a specific environment we need 3 files: configuration, genesis, and topology. 

@disassembler already pointed me to Staging and Testnet configurations (https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/staging-config.nix) that your team maintains.

What about the `genesis` and `topology` files for Staging and Testnet environments - are they the same for all 3 environments or there are differences too?

**Comment by oneEdoubleD** [comment]
> Should we close this PR and use the configurations provided/maintained by devops?

A community member asked where they could access Testnet/Staging configuration files too, so I think we should still have them in the `cardano-node` repo. It's a lot more accessible for users and they shouldn't change that often such that it's an issue to maintain (especially if we use them for testing on Windows).

**Comment by jbgi** [comment]
> What about the `genesis` and `topology` files for Staging and Testnet environments - are they the same for all 3 environments or there are differences too?

Genesis are the `<staging|testnet|mainnet|>-genesis.json` files in https://github.com/input-output-hk/iohk-nix/tree/master/cardano-lib

As for the topology used by the scripts, it is generated, by default using the DNS A entry that gather all edge (aka relay) nodes of the cluster. Those dns names can be found here:
https://github.com/input-output-hk/iohk-nix/blob/dfce0e88fee7d5c29fd6f128d02be3594ae39c88/cardano-lib/default.nix#L56
(`relaysNew` attributes)

File: configuration/defaults/testnet/configuration.yaml (hunk 1 above)
Comment (dorin100) [reply configuration/defaults/testnet/configuration.yaml]: this should be fixed by the latest commit

File: configuration/defaults/staging/configuration.yaml (hunk 2 above)
Comment (dorin100) [reply configuration/defaults/staging/configuration.yaml]: this should be fixed by the latest commit

File: configuration/defaults/staging/configuration.yaml (hunk 3 above)
Comment (dorin100) [reply configuration/defaults/staging/configuration.yaml]: this should be fixed by the latest commit

**Comment by dorin100** [comment]
@oneEdoubleD , @ArturWieczorek - I updated the Staging and Testnet working configurations (I tested them on Windows). Please take a look and Merge the PR if everything is ok. Thank you.

**Comment by ArturWieczorek** [comment]
@dorin100 Due to folder renaming for **mainnet** from `configuration/defaults/mainnet/configuration.yaml`  to → `configuration/defaults/byron_mainnet/configuration.yaml` some scripts might be affected. 

There is a **mainnet.sh** script that will start and run node against mainnet. It is located here: https://github.com/input-output-hk/cardano-node/blob/master/scripts/mainnet.sh

After folder renaming it will not work - in order to fix this you need to change line no 39 from:
`run_node --config-name 'mainnet' \`
to:
`run_node --config-name 'byron_mainnet' \`

As configs for `staging` and `testnet` are new and were just added in this PR I think they should not affect anything.

**Comment by dorin100** [comment]
Good cath @ArturWieczorek . I had found that also `scripts/lib-node.sh` was using the old `mainnet` naming. Updated both of them in a new commit.

**Pull request merged**